### DISCUSSION TITLE
How to customize index table query ?

### DISCUSSION THREAD

**Discussion by maherelgamil** [filamentphp/filament#1451] [Help]
If you need to override or customize index query you have two ways: 

- Override function `getTableQuery` in `ListRecords` that's located in `Pages` folder inside your resource folder

```php
protected function getTableQuery(): Builder
{
    return static::getResource()::getEloquentQuery()->where('role', 'admin');
}
```

- Override function `getEloquentQuery` in your resource main file

```php

public static function getEloquentQuery(): Builder
{
    return static::getModel()::query()->where('role', 'admin');
}

```

But as @ryangjchandler mention, "overwriting getEloquentQuery could have unintended side effects when the query is used elsewhere".

**Comment by ryangjchandler** [comment]
@maherelgamil You should overwrite `getTableQuery` really - overwriting `getEloquentQuery` could have unintended side effects when the query is used elsewhere.

**Reply by maherelgamil** [reply]
@ryangjchandler You right but I mean all available options/ways to customize the query

**Comment by ramiropiro** [comment]
thank you very helpfull, solved my problem using getTableQuery instead of getEloquentQuery
